mypyc/genops.py

```diff
@@ -1677,8 +1677,14 @@ def get_assignment_target(self, lvalue: Lvalue,
                                                          reassign=False)
 
                     if self.fn_info.contains_nested and self.is_free_variable(symbol):
-                        return self.add_var_to_env_class(symbol, self.node_type(lvalue),
-                                                         self.fn_info, reassign=False)
+                        if self.fn_info.is_nested:
```

Comment:
How about grabbing the class with `env = self.fn_info.callable_class if self.fn_info.is_nested else self.fn_info` and then unconditionally passing that to the method call?